presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java

```diff
@@ -248,57 +268,69 @@ private static ConnectorTableMetadata getTableMetadata(String schemaName, TpchTa
     @Override
     public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint<ColumnHandle> constraint)
     {
-        TpchTableHandle table = (TpchTableHandle) tableHandle;
-        return new TableStatistics(new Estimate(getRowCount(table, Optional.of(constraint.getSummary()))), ImmutableMap.of());
+        TpchTableHandle tpchTableHandle = (TpchTableHandle) tableHandle;
+        String tableName = tpchTableHandle.getTableName();
+        TpchTable<?> tpchTable = TpchTable.getTable(tableName);
+        Predicate<Map<TpchColumn<?>, Object>> predicate = adaptPredicate(constraint);
+        TableStatisticsData tableStatisticsData = statisticsEstimator.estimateStats(tpchTable, predicate, tpchTableHandle.getScaleFactor());
+        return toTableStatistics(tableStatisticsData, tpchTableHandle, getColumnHandles(session, tpchTableHandle));
     }
 
-    private long getRowCount(TpchTableHandle tpchTableHandle, Optional<TupleDomain<ColumnHandle>> predicate)
+    private Predicate<Map<TpchColumn<?>, Object>> adaptPredicate(Constraint<ColumnHandle> constraint)
```

Comment:
Ended up passing a `Map<TpchColumn<?>, List<Object>> columnValuesRestrictions` to `estimateStatistics`, this map either is empty (no restrictions = TupleDomain.all()), or has all columns as keys and `emptyList()` as all values (total restriciton = TupleDomain.none()), or has one key `ORDER_STATUS` with the allowed values extracted from the tuple domain.

Not using a `Multimap` here because with it there's no way to distinguish between no entry for a key and an entry with value of an empty list.